cpp/test/util/device_atomics.cu

```diff
@@ -51,8 +51,8 @@ TEST(Raft, AtomicIncWarp)
   // Write all 1M thread indices to a unique location in `out_device`
   test_atomic_inc_warp_kernel<<<num_blocks, threads_per_block, 0, s>>>(counter.data(),
                                                                        out_device.data());
-
   // Copy data to host
+  RAFT_CUDA_TRY(cudaStreamSynchronize(s)); //ensure stream is done before copy
```

Comment:
Could we use a cudaMemcpyAsync below instead?

Reply:
Yes we can, and done.

Reply:
Please use raft::copy and raft::update_host instead of calling cuda directly.

Reply:
There are no existing raft calls ( or handles ) in this test. The original version was calling `cudaMemcpy` on memory being written to by a kernel on a different stream.